core/src/main/java/org/elasticsearch/index/query/ExistsQueryBuilder.java

```diff
@@ -134,7 +134,7 @@ public static Query newFilter(QueryShardContext context, String fieldPattern) {
                 (FieldNamesFieldMapper.FieldNamesFieldType)context.getMapperService().fullName(FieldNamesFieldMapper.NAME);
         if (fieldNamesFieldType == null) {
             // can only happen when no types exist, so no docs exist either
-            return Queries.newMatchNoDocsQuery();
+            return Queries.newMatchNoDocsQuery("Missing types in " + NAME);
```

Comment:
same here, "Missing types in exists", maybe beter sth. like "Missing types in 'exists' query" or something like that.